Lib/asyncio/base_events.py

```diff
@@ -795,6 +801,49 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
 
         return transport, protocol
 
+    async def start_tls(self, transport, protocol, sslcontext, *,
+                        server_side=False,
+                        ssl_handshake_timeout=None):
+        """Upgrade transport to TLS.
+
+        Return a new transport that *protocol* should start using
+        immediately.
+        """
+        if ssl is None:
+            raise RuntimeError('Python ssl module is not available')
+
+        if not isinstance(sslcontext, ssl.SSLContext):
+            raise TypeError(
+                f'sslcontext is expected to be an instance of ssl.SSLContext, '
+                f'got {sslcontext!r}')
+
+        if not getattr(transport, '_start_tls_compatible', False):
+            raise TypeError(
+                f'transport {self!r} is not supported by start_tls()')
+
+        if transport.get_write_buffer_size():
+            if transport._buffer_drained_fut is not None:
+                raise RuntimeError(
+                    f'cannot start_tls(); another operation is awaiting '
+                    f'for the write buffer to drain')
+
+            waiter = self.create_future()
+            transport._buffer_drained_fut = waiter
+            await waiter
+
+            assert transport._buffer_drained_fut is None
+
+        waiter = self.create_future()
+        app_transport = self._make_ssl_transport(
+            transport._sock, protocol, sslcontext, waiter,
```

Comment:
Perhaps use `transport.get_extra_info('socket')`?

Reply:
Since asyncio loop doesn't work uvloop transport (it's true for any loop implementation) using private attributes is perfectly fine from my perspective.